s3/src/main/scala/geotrellis/store/s3/S3LayerDeleter.scala

```diff
@@ -51,17 +52,20 @@ class S3LayerDeleter(
         iter
           .map { s3obj => ObjectIdentifier.builder.key(s3obj.key).build() }
           .toList
```

Comment:
Hmm, I think it should no longer be casted into the list, since after grouping we're triggering iterator via `foreach`. So it can be even smth like:

```scala
iter
  .map { s3obj => ObjectIdentifier.builder.key(s3obj.key).build() }
  .grouped(requestLimit).foreach { objIdentifiersChunk =>
    // ... code to send request to S3
   }

```